scene-lib/src/scenelib/annotations/el/AMethod.java

```diff
@@ -76,6 +101,98 @@ public String getMethodName() {
         return methodSignature.substring(0, methodSignature.indexOf("("));
     }
 
+    /**
+     * Get the type parameters of this method.
+     *
+     * @return the list of type parameters
+     */
+    public List<? extends TypeParameterElement> getTypeParameters() {
+        return typeParameters;
+    }
+
+    /**
+     * Set the type parameters of this method.
+     *
+     * @param the list of type parameters
+     */
+    public void setTypeParameters(List<? extends TypeParameterElement> typeParameters) {
+        if (typeParameters == null) {
+            return;
+        }
+        if (this.typeParameters != null && !this.typeParameters.equals(typeParameters)) {
+            throw new Error(String.format("setTypeParameters(%s): already is %s%n",
+                                          typeParameters, this.typeParameters));
+        }
+        this.typeParameters = typeParameters;
+    }
+
+    /**
+     * Populates the method parameter map for the method. This is called from the constructor, so
+     * that the method parameter map always has an entry for each parameter.
```

Comment:
This comment is no longer accurate - it was true of `AMethodWrapper`, but I think it is only called from `setFieldsFromMethodElement` now.